src/software/new_geom/polygon.cpp

```diff
@@ -2,9 +2,33 @@
 
 #include <unordered_set>
 
-Polygon::Polygon(const std::vector<Point>& points) : points_(points) {}
+Polygon::Polygon(const std::vector<Point>& points)
+    : points_(points), segments_(initSegments(points_))
+{
+    // we pre-compute the segments_ in the constructor to improve performance
+}
 
-Polygon::Polygon(const std::initializer_list<Point>& points) : points_(points) {}
+Polygon::Polygon(const std::initializer_list<Point>& points)
+    : Polygon(std::vector(points))
+{
+}
+
+std::vector<Segment> Polygon::initSegments(std::vector<Point> points)
```

Comment:
Might want to pass `point` by const reference here, especially given the concerns around performance.

Reply:
tried it, it doesn't improve performance